CesiumGltf/include/CesiumGltf/Ktx2TranscodeTargets.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include "CesiumGltf/Library.h"
+
+#include <optional>
+
+namespace CesiumGltf {
+
+/**
+ * @brief Supported gpu compressed pixel formats.
+ */
+enum CESIUMGLTF_API GpuCompressedPixelFormat {
+  ETC1_RGB,
+  ETC2_RGBA,
+  BC1_RGB,
+  BC3_RGBA,
+  BC4_R,
+  BC5_RG,
+  BC7_RGBA,
+  PVRTC1_4_RGB,
+  PVRTC1_4_RGBA,
+  ASTC_4x4_RGBA,
+  PVRTC2_4_RGB,
+  PVRTC2_4_RGBA,
+  ETC2_EAC_R11,
+  ETC2_EAC_RG11
+};
+
+/**
+ * @brief Represents the status of support for all
+ * {@link GpuCompressedPixelFormat}s on a client platform.
+ *
+ * Clients can use this struct to convey which gpu compressed pixel formats are
+ * supported. This can be used to construct a {@link Ktx2TranscodeTargets}.
+ *
+ * Each entry in the struct is a bool that represents whether the gpu
+ * compressed pixel format with the corresponding name is supported.
+ */
+struct CESIUMGLTF_API SupportedGpuCompressedPixelFormats {
+  bool ETC1_RGB;
```

Comment:
These would default to false, no? Or is it better for clarity to explicitly initialize to false?

Reply:
No, I don't think they're guaranteed to be initialized to false. In particular if you use it like this:

```
SupportedGpuCompressedPixelFormats supported;
...
```

Then fields of primitive type will have an undefined value.

But I'm not 100% sure because [Initialization in C++ is Bonkers](https://accu.org/journals/overload/25/139/brand_2379/). Best advice is to always initialize. You can do so here just by adding `{}` after each field:

```
bool ETC1_RGB{};
```